src/arch/MovieTexture/MovieTexture_FFMpeg.cpp

```diff
@@ -12,6 +12,8 @@
 #include <cstdint>
 #include <thread>
 
+std::mutex MovieDecoder_FFMpeg::m_avcodec_mutex;
```

Comment:
Prefer putting this in an anonymous namespace in this file instead declaring it as a static variable as a member of the class.

Also prefer anonymous namesapces over defining static globals going forward. C++ style guide says either is fine, but my general preference is for anonymous namespaces: https://google.github.io/styleguide/cppguide.html#Internal_Linkage

```cpp
namespace {

std::mutex m_avcodec_mutex;

}  // namespace
```

Additionally for things that aren't trivially destructible (like std::mutex), you can define a function-local static pointer (or reference) and return it (under common practices here: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables)

```cpp
namespace {

std::mutex* GetLock() {
  static const std::mutex* m_avcodec_mutex = new std::mutex();
  return m_avcodec_mutex;
}

}  // namespace
```

Reply:
I also strongly prefer anonymous namespaces so I am happy to accept this.